homeassistant/helpers/config_validation.py

```diff
@@ -1014,6 +1014,8 @@ def make_entity_service_schema(
         vol.All(
             vol.Schema(
                 {
+                    # The frontend stores data here. Don't use in core.
+                    vol.Remove("metadata"): dict,
```

Comment:
> I like this idea but is this a breaking change now? 

No, we have picked `metadata` at this level, because it wasn't in use yet 😉 
From that perspective, this is safe at this point.

Reply:
> What if a service accepted metadata as part of data and it had nothing to do with UI metadata?

This is possible, but the fix makes sure that works as expected; we only remove `metadata` if the service does not accept a `metdata` parameter.